com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/MyTextDocumentService.java

```diff
@@ -213,6 +223,28 @@ private void registerEngineAndAnalyze(String uri, String languageType, String te
     }
   }
 
+  /**
+   * This method
+   * @param uri
+   */
+  private void deleteDependencyFile(String uri) {
+    Path folderPath = null;
+    if (checkIfInitWasCalled()) {
+      folderPath =
+          Paths.get(
+              getDependencyFileFolderPath()
+                  + FileSystems.getDefault().getSeparator()
+                  + getFileNameFromURI(uri)
+                  + DEP_EXTENSION);
+    }
+    if (isFileExists(folderPath))
+      try {
+        Files.delete(folderPath);
+      } catch (IOException e) {
+        log.error(e.getMessage());
+      }
+  }
+
```

Comment:
It is definitely not the responsibility of this class. I think that `TextDocumentService` should just fire an event to make `CopybookService` to remove the dep file

Reply:
Wait, ```CopybookService``` lose this responsability, there is a ```CopybookDependencyService``` that is responsible to that. Why we need to propagate this event on the databus?

Reply:
Yes, it should be on `CopybookDependencyService`. 
The databus event is needed to remove a direct dependency between the `TextDocumentService` and the dependency service.

Reply:
There is no direct dependency between those two services, @temanbrcom , how do you see the dependency?

Reply:
There is no dependency because now TextDocumentService is applying file operations, but it shouldn't. It should just notify the `CopybookDependencyService` to delete dep files

Reply:
So, there is no dependency, I will take a look and see how I can notify the `CopybookDependencyService`

Reply:
To be clear - saying "remove" I meant "avoid"